tiproxy: update performance test and metrics (#16694)
pingcap/docs: tiproxy/tiproxy-performance-test.md

### Test results

| vCPU | Threads | QPS     | Avg latency (ms) | P95 latency (ms) | TiProxy CPU usage | TiDB overall CPU usage |
|------|---------|---------|------------------|------------------|-------------------|------------------------|
| 2    | 40      | 58508   | 0.68             | 0.97             | 190%              | 1200%                  |
| 4    | 80      | 104890  | 0.76             | 1.16             | 390%              | 2000%                  |
| 6    | 120     | 155520  | 0.77             | 1.14             | 590%              | 2900%                  |
| 8    | 160     | 202134  | 0.79             | 1.18             | 800%              | 3900%                  |

## Long connection test

### Test plan

This test aims to verify that a large number of idle connections have minimal impact on the QPS when the client uses long connections. This test creates 5000, 10000, and 15000 idle long connections, and then executes `sysbench`.

> In this test, the `conn-buffer-size` configuration is kept at the default value:

Reviewer: ```suggestion
This test uses the default value for the `conn-buffer-size` configuration:
```

```yaml
proxy.conn-buffer-size: 32768
```

> The following command is used to perform the test:

Reviewer: ```suggestion
Use the following command to perform the test:
```

    --db-driver=mysql \
    --mysql-db=sbtest \
    --mysql-host=$host \
    --mysql-port=$port \
    run --tables=32 --table-size=1000000
```

### Test results

| Connection count | QPS   | Avg latency (ms) | P95 latency (ms) | TiProxy CPU usage | TiProxy memory usage (MB) | TiDB overall CPU usage |
|------------------|-------|------------------|------------------|-------------------|---------------------------|------------------------|
| 5000             | 96620 | 0.52             | 0.64             | 330%              | 920                       | 1800%                  |
| 10000            | 96143 | 0.52             | 0.65             | 330%              | 1710                      | 1800%                  |
| 15000            | 96048 | 0.52             | 0.65             | 330%              | 2570                      | 1900%                  |

## Short connection test

### Test plan

This test aims to verify that frequent creation and destruction of connections have minimal impact on the QPS when the client uses short connections. This test starts another client program to create and disconnect 100, 200, and 300 short connections per second while executing `sysbench`.

> The following command is used to perform the test:

Reviewer: ```suggestion
Use the following command to perform the test:
```
